process/transaction/interceptedTransaction.go

```diff
@@ -364,6 +381,32 @@ func (inTx *InterceptedTransaction) verifySig(tx *transaction.Transaction) error
 	return inTx.singleSigner.Verify(senderPubKey, txMessageForSigVerification, tx.Signature)
 }
 
+// VerifyGuardianSig verifies if the guardian signature is valid
+func (inTx *InterceptedTransaction) VerifyGuardianSig(tx *transaction.Transaction) error {
+	txMessageForSigVerification, err := inTx.getTxMessageForGivenTx(tx)
+	if err != nil {
+		return err
+	}
+
+	if !inTx.txVersionChecker.IsGuardedTransaction(tx) {
+		if len(tx.GetGuardianAddr()) > 0 {
+			return fmt.Errorf("%w without guardian signature", process.ErrGuardianAddressNotExpected)
```

Comment:
might return the errors directly here and on L396 as they are describing what they do. 
Otherwise we would have something like: "guardian signature not expected without guardian signature"

Reply:
done